### PR TITLE
feat: add codegen config related apis

### PR DESCRIPTION
**Pull request by phani-srikar**
#### Description of changes

Adds new API to read parts of the codegen configuration like the path to generated GraphQL documents and frontend types.
[Click to view the API review doc](https://quip-amazon.com/Zsf4AUVFyQlY/API-review-Amplify-Codegen-Configuration-API)
This will enable the customers to programmatically render the imports for generated GraphQL documents and types in their application.

#### Description of how you validated changes
Added unit test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/main/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
- [ ] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-codegen/src/codegen-config/AmplifyCodeGenConfig.js [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
         if (!withoutInit) {
-          projectRoot = amplify.getEnvInfo().projectPath || process.cwd();
+          projectRoot = projectPath || process.cwd();
         } else {
           projectRoot = process.cwd();
         }
-        const configPath = join(projectRoot, '.graphqlconfig.yml');
-        this.gqlConfig = new graphQLConfig.GraphQLConfig(null, configPath);
-        this.gqlConfig.config = {};
+        this.gqlConfig = graphQLConfig.getGraphQLConfig(projectRoot);
```
Comment (AaronZyLee) [inline packages/amplify-codegen/src/codegen-config/AmplifyCodeGenConfig.js:21]: Not very familiar with the `graphql-config`. Does this change of method call have the same effect as the previous one?

File: packages/amplify-codegen/src/codegen-config/index.js [hunk 2]
```diff
@@ -1,12 +1,69 @@
...
+    const projectPath = context.amplify.getEnvInfo().projectPath;
+    config = new AmplifyCodeGenConfig(projectPath, withoutInit);
   }
   return config;
 }
 
-module.exports = loadConfig;
+function getCodegenConfig(projectPath) {
+  if (!projectPath) {
+    throw new Error('Invalid projectPath provided to get codegen configuration');
```
Comment (AaronZyLee) [inline packages/amplify-codegen/src/codegen-config/index.js:17]: nit: The `project path` can be used instead of `projectPath` for the error string part. Same as below.

File: packages/amplify-codegen-e2e-core/src/init/initProjectHelper.ts [hunk 3]
```diff
@@ -108,8 +108,7 @@ export function initAndroidProjectWithProfile(cwd: string, settings: Object): Pr
       .wait('Choose your default editor:')
       .sendLine(s.editor)
       .wait("Choose the type of app that you're building")
-      .send('j')
-      .sendCarriageReturn()
+      .sendLine('android')
```
Comment (AaronZyLee) [inline packages/amplify-codegen-e2e-core/src/init/initProjectHelper.ts:111]: Is this related to the CLI version update?

File: packages/amplify-codegen/src/codegen-config/index.js [hunk 4]
```diff
@@ -1,12 +1,69 @@
...
+  if (!projects.length > 0) {
+    throw new Error(`No projects were found in fetched codegen configuration: ${codegenConfig}`);
+  }
+
+  const cfg = projects[0];
+  const includeFiles = cfg.includes[0];
+
+  const opsGenDirectory = cfg.amplifyExtension.docsFilePath
+    ? cfg.amplifyExtension.docsFilePath
+    : path.dirname(path.dirname(includeFiles));
```
Comment (AaronZyLee) [inline packages/amplify-codegen/src/codegen-config/index.js:43]: Just curious on the case of docFilePath not being defined. Do we have a walkthrough case ending up in this situation? I look into the [code](https://github.com/aws-amplify/amplify-codegen/blob/f96e002435cc1c4c31d199cd4ffdca17f0856f59/packages/amplify-codegen/src/walkthrough/configure.js#L58) and do not find any unless the `.graphqlconfig` is manually edited.

Comment (AaronZyLee) [reply packages/amplify-codegen/src/codegen-config/index.js:43]: If it happens on manual edit only, I think we also need to check the `includeFiles` is in the correct form of glob expression ending up with `**/.*graphql`

**Review by AaronZyLee — CHANGES_REQUESTED** [review]

File: packages/amplify-codegen-e2e-core/src/init/initProjectHelper.ts (hunk 3 above)
Comment (phani-srikar) [reply packages/amplify-codegen-e2e-core/src/init/initProjectHelper.ts:111]: Yeah I think so. Bunch of e2es started failing post release, have fixed most.

File: packages/amplify-codegen/src/codegen-config/index.js (hunk 4 above)
Comment (phani-srikar) [reply packages/amplify-codegen/src/codegen-config/index.js:43]: I kind of took the same logic we use for [statements generation](https://github.com/aws-amplify/amplify-codegen/blob/f96e002435cc1c4c31d199cd4ffdca17f0856f59/packages/amplify-codegen/src/commands/statements.js#L39) to compute the destination.

**Review by AaronZyLee — APPROVED** [review]

**Review by dpilch — APPROVED** [review]
